src/Compilers/CSharp/Portable/Symbols/Source/TypeParameterConstraintClause.cs

```diff
@@ -55,6 +56,18 @@ internal enum TypeParameterConstraintKind
         AllNonNullableKinds = ReferenceType | ValueType | Constructor | Unmanaged,
     }
 
+    internal sealed class TypeParameterConstraintClauses
+    {
+        public readonly ImmutableArray<TypeParameterConstraintClause> TypeParameterConstraints;
+        public readonly bool UsedLightweightTypeConstraintBinding;
+
+        public TypeParameterConstraintClauses(ImmutableArray<TypeParameterConstraintClause> typeParameterConstraints, bool usedLightweightTypeConstraintBinding)
```

Comment:
>public TypeParameterConstraintClauses [](start = 8, length = 37)

Consider making the constructor `private` and using a `Create` method instead, to re-use instances for empty sets.